Formula/pipgrip.rb

```diff
@@ -53,6 +54,23 @@ class Pipgrip < Formula
 
   def install
     virtualenv_install_with_resources
+
+    # Create a symlink with Python version appended for Homebrew internal use
+    python3 = "python3.11"
+    pyver = Language::Python.major_minor_version python3
+    (libexec/"bin").install_symlink libexec/"bin/pipgrip" => "pipgrip-#{pyver}"
+
+    # Create env scripts using older Pythons for Homebrew internal use
+    env = { PYTHONPATH: libexec/Language::Python.site_packages(python3) }
+    Formula["six"].pythons.each do |python|
+      other_python = python.opt_libexec/"bin/python"
+      other_pyver = Language::Python.major_minor_version other_python
+      next if pyver == other_pyver
+
+      pipgrip = libexec/"bin/pipgrip-#{other_pyver}"
+      pipgrip.write_env_script other_python, libexec/"bin/pipgrip", env
+      chmod "+x", pipgrip
+    end
```

Comment:
Safer option would be multiple venvs in case Python 3.11-installed packages end up incompatible with Python 3.10.